frontend/src/metabase/setup/setup.unit.spec.tsx

```diff
@@ -0,0 +1,156 @@
+import { fireEvent, waitFor, within } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import fetchMock from "fetch-mock";
+import type { UsageReason } from "metabase-types/api";
+import {
+  createMockSettingsState,
+  createMockSetupState,
+  createMockState,
+} from "metabase-types/store/mocks";
+import { renderWithProviders, screen } from "__support__/ui";
+import { Setup } from "./components/Setup";
+
+const setup = async ({ step = 0 } = {}) => {
+  const state = createMockState({
+    setup: createMockSetupState({
+      step,
+    }),
+    settings: createMockSettingsState({
+      "available-locales": [["en", "English"]],
+    }),
+  });
+
+  fetchMock.post("path:/api/util/password_check", { valid: true });
+
+  renderWithProviders(<Setup />, { storeInitialState: state });
+
+  // there is some async stuff going on with the locale loading
+  await screen.findByText("Let's get started");
+
+  return;
+};
+
+describe("setup", () => {
+  describe("default step order", () => {
+    // eslint-disable-next-line jest/expect-expect
+    it.each([
+      ["What's your preferred language?", "1"],
```

Comment:
this test is just a sanity check put in place to check that I won't break random things when I'll do the big refactor